pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -494,10 +494,18 @@ func doInstallVolumeRef(ctx *zedmanagerContext, config types.AppInstanceConfig,
 	status *types.AppInstanceStatus, vrs *types.VolumeRefStatus) bool {
 
 	changed := false
+	var vrc *types.VolumeRefConfig
+	for i := range config.VolumeRefConfigList {
+		vrc = &config.VolumeRefConfigList[i]
+		if vrc.VolumeID == vrs.VolumeID {
```

Comment:
And if not found, then we take last random config? Not nice

Reply:
... and hopefully the list is not empty at all, otherwise null. Can you please be more conservative and add a check here? or if you are confident that this can't happen better assert.

Reply:
good catch, thank you!